src/coreclr/jit/valuenum.cpp

```diff
@@ -4297,7 +4307,10 @@ void ValueNumStore::GetConstantBoundInfo(ValueNum vn, ConstantBoundInfo* info)
 //           This enables optCreateJTrueBoundAssertion to immediatly create an OAK_NO_THROW
 //           assertion instead of the OAK_EQUAL/NOT_EQUAL assertions created by signed compares
 //           (IsVNCompareCheckedBound, IsVNCompareCheckedBoundArith) that require further processing.
-
+//
+// Note:
+//   For comparisons of the form constant <= length, this returns them as (constant - 1) < length 
```

Comment:
Not too sure what the thoughts about this are. It makes everything very simple and is true, but we're also changing how the relop is represented.